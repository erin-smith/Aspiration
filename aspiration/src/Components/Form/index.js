import React from "react";

function Form({ query, handleInputChange, handleFormSubmit }) {
  return (
    <form>
      <div className="form-group">
        <label htmlFor="Query">
          <strong>Topic</strong>
        </label>
        <input
          className="form-control"
          id="Topic"
          type="text"
          value={query}
          placeholder="html"
          name="q"
          onChange={handleInputChange}
          required
        />
      </div>
      <div className="pull-right">
        <button
          onClick={handleFormSubmit}
          type="submit"
          className="btn btn-lg btn-info float-right"
        >
          Search
        </button>
      </div>
    </form>
  );
}

export default Form;
