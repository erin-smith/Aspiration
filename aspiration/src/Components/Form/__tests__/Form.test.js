import * as React from "react";
import Form from "../index";
import renderer from "react-test-renderer";

describe("Form", () => {
  test("Should match the snapshot", () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
