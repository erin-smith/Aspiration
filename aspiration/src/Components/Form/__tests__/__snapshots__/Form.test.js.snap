// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match the snapshot 1`] = `
<form
  className="search"
>
  <div
    className="form-group"
  >
    <label
      htmlFor="Query"
    >
      <strong>
        Topic
      </strong>
    </label>
    <input
      className="form-control"
      name="topic"
      placeholder="html"
      required={true}
      type="text"
    />
  </div>
  <div
    className="pull-right"
  >
    <button
      className="btn btn-lg btn-info float-right"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
